dispatch/route_test.go

```diff
@@ -383,3 +383,238 @@ routes:
 	require.Equal(t, child1.RouteOpts.GroupByAll, true)
 	require.Equal(t, child2.RouteOpts.GroupByAll, false)
 }
+
+func TestRouteMatchers(t *testing.T) {
```

Comment:
Perhaps also add a test that tries out a combination of new-style and old-style matchers? Essentially just duplicate this test and write some of the matchers in the old YAML syntax.